.github/workflows/ci.yml

```diff
@@ -129,41 +175,59 @@ jobs:
+          echo "Applying patch: $patch_file"
+          git apply "$patch_file"
+        done
         # Build
         mkdir build
         cd build
-        if [[ "${cling_on^^}" == "ON" ]]; then
+        if [[ "${cling_on}" == "ON" ]]; then
           cmake -DLLVM_ENABLE_PROJECTS=clang                  \
                 -DLLVM_EXTERNAL_PROJECTS=cling                \
                 -DLLVM_EXTERNAL_CLING_SOURCE_DIR=../../cling  \
                 -DLLVM_TARGETS_TO_BUILD="host;NVPTX"          \
                 -DCMAKE_BUILD_TYPE=Release                    \
                 -DLLVM_ENABLE_ASSERTIONS=ON                   \
-                -DLLVM_USE_LINKER=lld                         \
                 -DCLANG_ENABLE_STATIC_ANALYZER=OFF            \
                 -DCLANG_ENABLE_ARCMT=OFF                      \
                 -DCLANG_ENABLE_FORMAT=OFF                     \
                 -DCLANG_ENABLE_BOOTSTRAP=OFF                  \
                 ../llvm
-          cmake --build . --target clang --parallel $(nproc --all)
-          cmake --build . --target cling --parallel $(nproc --all)
-          # Now build gtest.a and gtest_main for CppInterOp to run its tests.
-          cmake --build . --target gtest_main --parallel $(nproc --all)
+          if [[ "${os}" == "macos*" ]]; then
+            cmake --build . --target clang --parallel $(sysctl -n hw.physicalcpu)
+            cmake --build . --target cling --parallel $(sysctl -n hw.physicalcpu)
+            # Now build gtest.a and gtest_main for CppInterOp to run its tests.
+            cmake --build . --target gtest_main --parallel $(sysctl -n hw.physicalcpu)
+          else
+            cmake --build . --target clang --parallel $(nproc --all)
+            cmake --build . --target cling --parallel $(nproc --all)
+            # Now build gtest.a and gtest_main for CppInterOp to run its tests.
+            cmake --build . --target gtest_main --parallel $(nproc --all)
+          fi          
         else
           cmake -DLLVM_ENABLE_PROJECTS=clang                  \
                 -DLLVM_TARGETS_TO_BUILD="host;NVPTX"          \
                 -DCMAKE_BUILD_TYPE=Release                    \
                 -DLLVM_ENABLE_ASSERTIONS=ON                   \
-                -DLLVM_USE_LINKER=lld                         \
                 -DCLANG_ENABLE_STATIC_ANALYZER=OFF            \
                 -DCLANG_ENABLE_ARCMT=OFF                      \
                 -DCLANG_ENABLE_FORMAT=OFF                     \
                 -DCLANG_ENABLE_BOOTSTRAP=OFF                  \
                 ../llvm
-          cmake --build . --target clang clang-repl --parallel $(nproc --all)
+          if [[ "${os}" == "macos*" ]]; then
+            cmake --build . --target clang clang-repl --parallel $(sysctl -n hw.physicalcpu)
+          else
+            cmake --build . --target clang clang-repl --parallel $(nproc --all)
+          fi
```

Comment:
Can you make this somewhere in the settings of the build and then just use ncpus instead of if-stmts everywhere?

```
if [[ "${os}" == "macos*" ]]; then
   ncpus=$(sysctl -n hw.physicalcpu)
else
   ncpus=$(nproc --all)
fi
```

Reply:
Maybe we should use hw.ncpu instead hw.physicalcpu

Reply:
- Made an environment variable to store number of cpu cores and replaced hw.physicalcpu to hw.ncpu